File: Samples/Verbum/how-to-embed-a-barcode-into-a-document-e20025-11.1.6-/VB/Extensions/BarCode/BarCodeGeneratorHelper.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports System.Collections.Generic
	Imports System.Linq
Namespace BizPad

	Friend NotInheritable Class BarCodeGeneratorHelper
		Private Sub New()
		End Sub
		Public Shared Function IsValidText(ByVal generator As IBarCodeGenerator, ByVal text As String) As Boolean
			If String.IsNullOrEmpty(text) Then
				Return False
			End If
			Dim charSet As String = generator.GetValidCharSet()
			Dim length As Integer = text.Length
			For i As Integer = 0 To length - 1
				If charSet.IndexOf(text.Chars(i)) < 0 Then
					Return False
				End If
			Next i
			Return True
		End Function
	End Class
End Namespace

File: Samples/Verbum/how-to-embed-a-barcode-into-a-document-e20025-11.1.6-/CS/Extensions/BarCode/BarCodeGeneratorHelper.cs
namespace BizPad {
    using System;
    using System.Collections.Generic;
    using System.Linq;

    internal static class BarCodeGeneratorHelper {
        public static bool IsValidText(IBarCodeGenerator generator, string text) {
            if (string.IsNullOrEmpty(text)) return false;
            string charSet = generator.GetValidCharSet();
            int length = text.Length;
            for (int i = 0; i < length; i++)
                if (charSet.IndexOf(text[i]) < 0)
                    return false;
            return true;
        }
    }
}
